tokio-trace-fmt/src/filter.rs

```diff
@@ -149,8 +150,19 @@ impl Filter for EnvFilter {
                     let in_span = ctx
                         .visit_spans(|_, span| {
                             if span.name() == desired {
-                                // Return `Err` to short-circuit the span visitation.
-                                Err(())
+                                if let Some((ref field, ref val)) = directive.in_field_value {
+                                    let fields = span.fields();
+                                    let matcher = format!("{}={:?}", field, val);
+
+                                    if fields == matcher {
```

Comment:
I think you want
```suggestion
                                    if fields.contains(matcher) {
```
If you're filtering for something like "foo=23", what you have now will only return true if the fields are _only_ "foo=23". Since a span might have more than one field, you also want to accept something like "foo=23 bar=false baz="whatever"".